Enhance OpenSCAP page (#938)
uyuni-project/uyuni-docs: modules/administration/pages/openscap.adoc

Before you begin, you need to prepare your client systems for SCAP scanning.

For traditional and Salt clients, install the OpenSCAP scanner and the SCAP Security Guide (content) packages before you begin. Depending on the operating system, these packages are included either on the base operating system, or in the {productname} Client Tools.

> The table below listst the packages you need depending on your client operating system:

Reviewer: ```suggestion
The table below lists the packages you need depending on your client operating system:
```

.OpenSCAP packages
[%header,cols=3]
|===
| Operating system | Scanner | Content
| SLES | openscap-utils | scap-security-guide
| openSUSE | openscap-utils | scap-security-guide
| RHEL | openscap-utils | scap-security-guide-redhat
| CentOS | openscap-utils | scap-security-guide-redhat
| Oracle Linux | openscap-utils | scap-security-guide-redhat
| Ubuntu | libopenscap8 | scap-security-guide-ubuntu
| Debian | libopenscap8 | scap-security-guide-debian
|===

> RHEL 7 and compatible provide a `scap-security-guide` package which contains outdated content. You are advised to use the `scap-security-guide-redhat` package you will find in the {productname} Client Tools.

Reviewer: ```suggestion
RHEL 7 and compatible systems provide a [package]``scap-security-guide`` package, which contains outdated contents.
You are advised to use the [package]``scap-security-guide-redhat`` package you will find in the {productname} Client Tools.
```

====

When you have created your content files, you need to transfer the file to the client.

> You can do this in the same way as you move any other file, using physical storage media, or across a network with Salt (e. g. https://docs.saltproject.io/en/latest/ref/cli/salt-cp.html[salt-cp] or the https://docs.saltproject.io/en/latest/ref/file_server/index.html[Salt File Server]), [command]``ftp`` or [command]``scp``.

Reviewer: ```suggestion
You can do this in the same way as you move any other file, using physical storage media, or across a network with Salt (for example, https://docs.saltproject.io/en/latest/ref/cli/salt-cp.html[salt-cp] or the https://docs.saltproject.io/en/latest/ref/file_server/index.html[Salt File Server]), [command]``ftp`` or [command]``scp``.
```

We recommend that you create a package to distribute content files to clients that you are managing with {productname}.
Packages can be signed and verified to ensure their integrity.
For more information, see xref:administration:custom-channels.adoc[].

== Find OpenSCAP profiles

Different operating systems make available different OpenSCAP content files and profiles. One content file may contain more than one profile.

On RPM-based operating systems, use this command to determine the location of the available SCAP files:

----

> $ rpm -ql <scap-security-guide-package-name-from-table>

Reviewer: ```suggestion
rpm -ql <scap-security-guide-package-name-from-table>
```

----

On DEB-based operating systems, use this command to determine the location of the available SCAP files:

----

> $ dpkg -L <scap-security-guide-package-name-from-table>

Reviewer: ```suggestion
dpkg -L <scap-security-guide-package-name-from-table>
```

----

> When you have identified one SCAP content file that suits your needs, list what are the available profiles on the client:

Reviewer: ```suggestion
When you have identified one SCAP content file that suits your needs, list profiles available on the client:
```

----

> $ oscap info ./usr/share/xml/scap/ssg/content/ssg-sle15-ds-1.2.xml

Reviewer: ```suggestion
oscap info ./usr/share/xml/scap/ssg/content/ssg-sle15-ds-1.2.xml
```

> Document type: Source Data Stream

Reviewer: That outputs:

----

Author: I don't know what you mean, it looks fine here :-?

![image](https://user-images.githubusercontent.com/50680111/117144005-b86e8700-adb1-11eb-8605-85b4f02e3e4f.png)

Reviewer: I wanted to separate the command from the output of the command (two blocks).  But your way is also fine.

                        https://ftp.suse.com/pub/projects/security/oval/suse.linux.enterprise.15.xml
                                system: http://oval.mitre.org/XMLSchema/oval-definitions-5
Checks:
        Ref-Id: scap_org.open-scap_cref_ssg-sle15-oval.xml
        Ref-Id: scap_org.open-scap_cref_ssg-sle15-ocil.xml
        Ref-Id: scap_org.open-scap_cref_ssg-sle15-cpe-oval.xml
Dictionaries:
        Ref-Id: scap_org.open-scap_cref_ssg-sle15-cpe-dictionary.xml
----

Take a note of the file paths and profiles for performing the scan.

== Perform an Audit Scan

When you have installed or transferred your content files, you can perform audit scans.
Audit scans can be triggered using the {productname} {webui}.
You can also use the {productname} API to schedule regular scans.

.Procedure: Running an Audit Scan from the {webui}
. In the {productname} {webui}, navigate to menu:Systems[Systems List] and select the client you want to scan.
. Navigate to the [guimenu]``Audit`` tab, and the [guimenu]``Schedule`` subtab.
. In the [guimenu]``Path to XCCDF Document`` field, enter the parameters for the SCAP template and profile you want to use on the client.
  For example:

>   ```

Reviewer: ```suggestion
----
```

  Command: /usr/bin/oscap xccdf eval
  Command-line arguments: --profile xccdf_org.ssgproject.content_profile_standard
  Path to XCCDF document: /usr/share/xml/scap/ssg/content/ssg-sle15-ds-1.2.xml

>   ```

Reviewer: ```suggestion
----

```
